src/main/java/com/aws/greengrass/provisioning/ProvisioningPluginFactory.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.provisioning;
+
+public class ProvisioningPluginFactory {
+    public <T> T getPluginInstance(Class pluginClass) throws IllegalAccessException, InstantiationException {
+        return (T)pluginClass.newInstance();
```

Comment:
this requires the plugin to have a default constructor. That's fine, but I just want to make sure you know

Reply:
Yes, have kept it this way for now. Working on refactoring it to use kernel context.

Reply:
This might be a good idea though, this way the plugin cannot get access to classes that we want to keep more hidden.

Reply:
Ya good point.